### PR TITLE
Added recipes to convert between XP ingots and blocks

### PR DESCRIPTION
**Pull request by pd93**
image
image

### CODE REVIEW

File: src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    protected void buildShapelessRecipes(Consumer<IFinishedRecipe> consumer) {
+        ShapelessRecipeBuilder.shapeless(ModItems.XP_INGOT.get(), 9)
+                .requires(ModBlocks.XP_BLOCK.get())
+                .unlockedBy("has_item", has(ModItems.XP_INGOT.get()))
+                .save(consumer);
+
+        ShapedRecipeBuilder.shaped(ModBlocks.XP_BLOCK.get())
```
Comment (iainvm) [inline src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java:24]: Should we move Shaped Recipes to somewhere else?

File: src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package org.hackedtogether.xpadditions.data;
+
+import net.minecraft.data.*;
```
Comment (iainvm) [inline src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java:3]: Don't import all of a package

**Review by iainvm — CHANGES_REQUESTED** [review]

File: src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java (hunk 2 above)
Comment (pd93) [reply src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java:3]: All of the package is used, so intellij automatically does this

File: src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java (hunk 1 above)
Comment (pd93) [reply src/main/java/org/hackedtogether/xpadditions/data/ModRecipeProvider.java:24]: There is no other method on `RecipeProvider` in which to place these. This seems like the only way to do it right now.

**Review by iainvm — APPROVED** [review]

**Pull request merged**